building-and-deploying-a-fastapi-app.md

```diff
@@ -0,0 +1,541 @@
+
+```
+curl -X PUT "http://0.0.0.0:8080/recipe?id=1" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{  \"name\": \"Buns\",}"
+```
+
+The UPDATE request returns this response:
+
+```json
+{
+  "message": "Recipe updated successfully."
+}
+```
+
+Let's define the final route, DELETE. The DELETE route is used to remove recipe in your application using their ID. Beneath the UPDATE route, add the following:
+
+```py
+@app.delete("/recipe/{id}", tags=["Recipe"])
+def delete_recipe(id: int) -> dict:
+    if id > len(recipes) or id < 1:
+        return {
+            "error": "Invalid ID passed"
+        }
+
+    for recipe in recipes:
+        if recipe['id'] == id:
+            recipes.remove(recipe)
+            return {
+                "message": "Recipe deleted successfully."
+            }
+
+    return {
+        "error": "No such recipe with ID {} exist".format(id)
+    }
+
+```
+
+In the code block above, you first check if the recipe whose ID is passed exists. If it does, the recipe is removed. Otherwise, an error message is returned. Test the DELETE route:
+
+```sh
+curl -X DELETE "http://0.0.0.0:8080/recipe/1" -H  "accept: application/json"
+```
+
+The request above return this response:
+
+```json
+{
+  "message": "Recipe deleted successfully."
+}
+```
+
+You have successfully built the CRUD application. 
+
+## Deploying to Okteto
+
+In this section, you will be deploying the application to Okteto. 
+
+Create an account on [Okteto](https://cloud.okteto.com) if you do not have one setup. Additionally, [Install the Okteto CLI](https://okteto.com/docs/getting-started#step-2-install-the-okteto-cli) on your machine.
+
+
+Before proceeding, create a _.gitignore_ file in the project folder to prevent checking in the "venv" folder to git:
+
+
+```
+$ touch .gitignore
+```
+
+Add the following:
+
+
+```
+venv
+__pycache__
+```
+
+With the necessary installations in place, log on to Okteto from your terminal using the command:
+
+```sh
+$ okteto login
+```
+
+After logging in, run `okteto init` to create an Okteto manifest file. The command automatically generates an `okteto.yml` file. Overwrite the content to:
+
+```yaml
+name: fastapi-okteto
+command: bash
+workdir: /app
+```
+
+The next step is to create a stack manifest file for Okteto. The manifest file removes the need to deal with the complexities of Kubernetes manifests.
+
+```
+$ touch okteto-stack.yaml
+```
+
+In the `okteto-stack.yaml` file, define the application using a docker-compose format:
+
+```yaml
+name: fastapi-crud
+services:
+  fastapi:
+    public: true
+    image: fastapi-crud:latest
+    build: .
+    replicas: 1
+    ports:
+      - 8080
+    resources:
+      cpu: 100m
+      memory: 128Mi
+```
+
+In the manifest file above, you defined your application's name and the services your app is made up of, fastapi. The fastapi service exposes the port `8080` and allocates itself with 100 mili CPUs and 128Mi of memory. 
```

Comment:
could you also explain that the app will be exposed on a public HTTPs endpoint, with a valid certificate?

Reply:
Sure :)